src/tests/class_tests/openms/source/TIC_test.cpp

```diff
@@ -0,0 +1,161 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Chris Bielow$
+// $Authors: Tom Waschischeck $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/CONCEPT/ClassTest.h>
+#include <OpenMS/test_config.h>
+
+///////////////////////////
+
+#include <OpenMS/QC/TIC.h>
+
+///////////////////////////
+
+START_TEST(TIC, "$Id$")
+
+/////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////
+using namespace OpenMS;
+using namespace std;
+
+TIC* ptr = nullptr;
+TIC* nullPointer = nullptr;
+START_SECTION(TIC())
+  ptr = new TIC();
+  TEST_NOT_EQUAL(ptr, nullPointer)
+END_SECTION
+
+START_SECTION(~TIC())
+  delete ptr;
+END_SECTION
+
+// Dummy peakmap
+PeakMap exp;
+exp.resize(4);
+Peak1D p;
+
+// MS spectrum at RT = 0
+p.setMZ(5.0);
+p.setIntensity(3);
+exp[0].push_back(p);
+p.setMZ(10.0);
+p.setIntensity(5);
+exp[0].push_back(p);
+exp[0].setMSLevel(1);
+exp[0].setRT(0);
+
+// MS spectrum at RT = 2
+p.setMZ(5.0);
+p.setIntensity(2);
+exp[1].push_back(p);
+exp[1].setMSLevel(1);
+exp[1].setRT(2);
+
+// MSMS spectrum at RT = 2
+p.setMZ(5.0);
+p.setIntensity(0.5);
+exp[2].push_back(p);
+exp[2].setMSLevel(2);
+exp[2].setRT(2);
+
+// MS spectrum at RT = 5
+p.setMZ(5.0);
+p.setIntensity(2.0);
+exp[3].push_back(p);
+p.setMZ(10.0);
+p.setIntensity(3.0);
+exp[3].push_back(p);
+p.setMZ(15.0);
+p.setIntensity(4.0);
+exp[3].push_back(p);
+exp[3].setMSLevel(1);
+exp[3].setRT(5);
+
+exp.updateRanges();
+
+START_SECTION(QCBase::Status requires() const)
+  TIC tic;
+  TEST_EQUAL((tic.requires() == QCBase::Status(QCBase::Requires::RAWMZML)),true);
+END_SECTION
+
+START_SECTION(void compute(const MSExperiment &exp, float bin_size) && vector<MSChromatogram> getResults() const)
+{
+  TIC tic;
+  TEST_EQUAL(tic.getResults().empty(),true);
+  tic.compute(exp); // no binning
+  tic.compute(exp,2.0); // bin size smaller than highest RT
+  tic.compute(exp,6.0); // bin size bigger than highest RT
+  tic.compute(exp,-1.0); // negative bin size
+
+  // empty MSExperiment
+  MSExperiment exp2;
+  tic.compute(exp2);
+
+  TEST_EQUAL(tic.getResults().size(),5);
```

Comment:
could be replace with `ABORT_IF(...)` (also in the next line) to avoid a segfault in the following lines